2214/2/battle_expert.rb

```diff
@@ -0,0 +1,40 @@
+require 'terminal-table'
+require_relative 'popular_words_counter'
+require_relative 'battler'
+require_relative 'battle'
+require_relative 'battles_by_name_giver'
+
+class BattleExpert
+  include BattlesByNameGiver
+  BATTLES_FOLDER = 'Battles'.freeze
+  def describe_battlers(top_bad_words)
+    puts Terminal::Table.new rows: sorted_battlers.first(top_bad_words).map(&:describe_yourself)
+  end
+
+  def find_popular_words(battler_name, top_words = 30)
+    if battlers_names.include? battler_name
+      PopularWordsCounter.new(battlers.find { |battler| battler.name.eql? battler_name }, top_words).count
+    else
+      puts "Я не знаю МЦ #{battler_name}. Зато мне известны:"
+      battlers_names.each { |battler| puts battler }
+    end
+  end
+
+  private
+
+  def sorted_battlers
+    battlers.sort_by(&:bad_words_per_round).reverse
```

Comment:
Imagine we call this method twice. Each time it will sort the array from scratch and this would be time-consuming. I suggest you using **memoization**.